File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {

        unordered_set<int> set;
        for(auto& num : nums) {
            set.insert(num);
        }

        int maxLength = 0;
        for (auto& num : set) {
            if (set.find(num - 1) != set.end()) {
                continue;
            }

            int curStreak = 1;
            int curNum = num;

            while (set.find(curNum + 1) != set.end()) {
                curStreak++;
                curNum++;
            }

            maxLength = max(maxLength, curStreak);
        }

        return maxLength;
        
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        maxStreak = 0
        s = set()

        for num in nums:
            s.add(num)

        for num in s:
            # Only start counting when num is the start of a sequence
            if num - 1 in s:
                continue
            
            curStreak = 1
            curNum = num

            while curNum + 1 in s:
                curStreak += 1
                curNum = curNum + 1
            
            maxStreak = max(maxStreak, curStreak)
            
        
        return maxStreak
        